src/software/ai/move_rule/move_rule_manager_test.cpp

```diff
@@ -0,0 +1,162 @@
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::DIRECT_FREE_THEM);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::INDIRECT_FREE_THEM);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::TIMEOUT_US);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::TIMEOUT_THEM);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::GOAL_US);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::GOAL_THEM);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::BALL_PLACEMENT_THEM);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+}
+
+TEST(MoveRuleTest, test_avoid_areas_from_game_start_or_us_states)
+{
+    GameState game_state;
+    MoveTactic move_tactic;
+    MoveRuleManager manager;
+    std::set<MoveRule> correct_move_rules;
```

Comment:
I'd suggest using a test fixture so that we don't need to re-declare these variables every time.

Reply:
Ok